newIDE/app/src/Utils/UpdateDiscordRichPresence.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import { useEffect, useState } from 'react';
+import optionalRequire from './OptionalRequire';
+const electron = optionalRequire('electron');
+const ipc = electron ? electron.ipcRenderer : null;
+
+const updateDiscordRichPresence = (project: ?gdProject) => {
+  if (ipc === null) return;
+  if (!updateDiscordRichPresence.date)
+    updateDiscordRichPresence.date = Date.now();
+
+  let state = {
+    details: project ? 'Working on:' : 'Not working on',
+    state: project ? project.getName() : 'any game',
+    startTimestamp: updateDiscordRichPresence.date,
+    largeImageKey: 'gdicon',
+    largeImageText: 'GDevelop',
+  };
+
+  ipc.send('update-discord-rich-presence', state);
```

Comment:
uber nitpicking: you call this object a "state", but you call it a "config" in `DiscordRichPresence.js`. Choose a term (probably the one used in Discord documentation) and stick to it :)